pkg/sql/schemachanger/scbuild/internal/scbuildstmt/create_index.go

```diff
@@ -103,20 +140,20 @@ func CreateIndex(b BuildCtx, n *tree.CreateIndex) {
 			// initial primary key and include the union of all of the added and
 			// dropped columns.
 			if target == scpb.ToPublic {
-				source = t
+				idxSpec.primary = t
 			}
 		}
 	})
 	if n.Unique {
-		index.ConstraintID = b.NextTableConstraintID(index.TableID)
+		idxSpec.secondary.ConstraintID = b.NextTableConstraintID(idxSpec.secondary.TableID)
```

Comment:
Shouldn't we be setting `idxSpec.secondary.IsUnique = true` here as well? I found out that we set it above when setting `idxSpec.secondary` but I think it would make more sense here.